contracts/vaults/GammaVault/RibbonGammaVault.sol

```diff
@@ -0,0 +1,996 @@
+        require(
+            newManagementFee < 100 * Vault.FEE_MULTIPLIER,
+            "Invalid management fee"
+        );
+
+        // We are dividing annualized management fee by num weeks in a year
+        uint256 tmpManagementFee =
+            newManagementFee.mul(Vault.FEE_MULTIPLIER).div(WEEKS_PER_YEAR);
+
+        emit ManagementFeeSet(managementFee, newManagementFee);
+
+        managementFee = tmpManagementFee;
+    }
+
+    /**
+     * @notice Sets the performance fee for the vault
+     * @param newPerformanceFee is the performance fee (6 decimals). ex: 20 * 10 ** 6 = 20%
+     */
+    function setPerformanceFee(uint256 newPerformanceFee) external onlyOwner {
+        require(
+            newPerformanceFee < 100 * Vault.FEE_MULTIPLIER,
+            "Invalid performance fee"
+        );
+
+        emit PerformanceFeeSet(performanceFee, newPerformanceFee);
+
+        performanceFee = newPerformanceFee;
+    }
+
+    /**
+     * @notice Sets a new cap for deposits
+     * @param newCap is the new cap for deposits
+     */
+    function setCap(uint256 newCap) external onlyOwner {
+        require(newCap > 0, "!newCap");
+        ShareMath.assertUint104(newCap);
+        emit CapSet(vaultParams.cap, newCap);
+        vaultParams.cap = uint104(newCap);
+    }
+
+    /************************************************
+     *  DEPOSIT & WITHDRAWALS
+     ***********************************************/
+
+    /**
+     * @notice Deposits the `asset` from msg.sender.
+     * @param amount is the amount of `asset` to deposit
+     */
+    function deposit(uint256 amount) external nonReentrant {
+        require(amount > 0, "!amount");
+
+        _depositFor(amount, msg.sender);
+
+        // An approve() by the msg.sender is required beforehand
+        IERC20(USDC).safeTransferFrom(msg.sender, address(this), amount);
+    }
+
+    /**
+     * @notice Deposits the `asset` from msg.sender added to `creditor`'s deposit.
+     * @notice Used for vault -> vault deposits on the user's behalf
+     * @param amount is the amount of `asset` to deposit
+     * @param creditor is the address that can claim/withdraw deposited amount
+     */
+    function depositFor(uint256 amount, address creditor)
+        external
+        nonReentrant
+    {
+        require(amount > 0, "!amount");
+        require(creditor != address(0));
+
+        _depositFor(amount, creditor);
+
+        // An approve() by the msg.sender is required beforehand
+        IERC20(USDC).safeTransferFrom(msg.sender, address(this), amount);
+    }
+
+    /**
+     * @notice Mints the vault shares to the creditor
+     * @param amount is the amount of `asset` deposited
+     * @param creditor is the address to receieve the deposit
+     */
+    function _depositFor(uint256 amount, address creditor) private {
+        uint256 currentRound = vaultState.round;
+        uint256 totalWithDepositedAmount = totalBalance().add(amount);
```

Comment:
General comment: Since this is a new contract altogether, we can remove the use of SafeMath from the vault contract. It's redundant because we are already using Solidity 0.8